test/tools/ossfuzz/protomutators/YulProtoMutator.h

```diff
@@ -0,0 +1,191 @@
+#pragma once
+
+#include <test/tools/ossfuzz/yulProto.pb.h>
+
+#include <libsolutil/Common.h>
+
+#include <src/libfuzzer/libfuzzer_macro.h>
+
+#include <random>
+
+namespace solidity::yul::test::yul_fuzzer
+{
+
+using ProtobufMessage = google::protobuf::Message;
+using ProtobufDesc = google::protobuf::Descriptor;
+
+template <typename Proto>
+using LPMPostProcessor = protobuf_mutator::libfuzzer::PostProcessorRegistration<Proto>;
+
+template <typename Proto>
+using FuzzMutatorCallback = std::function<void(Proto*, unsigned int)>;
+
+template <typename P>
+struct YulProtoCBRegistration
+{
+	YulProtoCBRegistration(FuzzMutatorCallback<P> const& _callback)
+	{
+		LPMPostProcessor<P> reg = {_callback};
+	}
+};
+
+class MutationInfo: public ScopeGuard
+{
+public:
+	MutationInfo(ProtobufMessage* _message, std::string const& _info);
+
+	static void print(std::string const& _str)
+	{
+		std::cout << _str << std::endl;
+	}
+	void exitInfo();
+
+	ProtobufMessage* m_protobufMsg;
+};
+
+struct YulRandomNumGenerator
+{
+	using RandomEngine = std::minstd_rand;
+
+	explicit YulRandomNumGenerator(unsigned _seed): m_random(RandomEngine(_seed)) {}
+
+	unsigned operator()()
+	{
+		return m_random();
+	}
+
+	RandomEngine m_random;
+};
+
+template <typename Proto>
+using CustomFuzzMutator = std::function<void(Proto*, YulRandomNumGenerator& _rand)>;
+
+struct YulProtoMutator
+{
+	enum class PrintChanges { Yes, No };
+
+	template <typename T>
+	static void functionWrapper(
+		CustomFuzzMutator<T> const& _callback,
+		T* _message,
+		unsigned int _seed,
+		unsigned _period,
+		std::string const& _info,
+		PrintChanges _printChanges = PrintChanges::No
+	);
+
+	/// Return an integer literal of the given value.
+	/// @param _value: Value of the integer literal
+	static Literal* intLiteral(unsigned _value);
+
+	/// Return a variable reference
+	/// @param _seed: Pseudo-random unsigned integer used as index
+	/// of variable to be referenced
+	static VarRef* varRef(unsigned _seed);
+
+	/// Return a literal expression
+	/// @param _value: value of literal expression
+	static Expression* litExpression(unsigned _value);
+
+	/// Return a reference expression
+	/// @param _seed: Pseudo-random unsigned integer used as index
+	/// of variable to be referenced
+	static Expression* refExpression(YulRandomNumGenerator& _rand);
```

Comment:
```suggestion
	static Expression* refExpression(YulRandomNumGenerator const& _rand);
```

Reply:
`_rand` cannot be `const` here because we need to invoke a non const function inside `YulRandomNumGenerator`